From 532d57de0963b0f6eba97cf5d5ad2d69437859a0 Mon Sep 17 00:00:00 2001
From: Pavel Zakharov <pavel.zakharov@delphix.com>
Date: Tue, 19 Jun 2018 13:14:27 -0400
Subject: [PATCH] 0017-Don-t-call-do_pam_setcred-twice.patch

---
 platform.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/platform.c b/platform.c
index 18c7751..f23a8c9 100644
--- a/platform.c
+++ b/platform.c
@@ -145,7 +145,7 @@ platform_setusercontext(struct passwd *pw)
 void
 platform_setusercontext_post_groups(struct passwd *pw)
 {
-#if !defined(HAVE_LOGIN_CAP) && defined(USE_PAM)
+#if !defined(HAVE_LOGIN_CAP) && defined(USE_PAM) && !defined(PAM_SUN_CODEBASE)
 	/*
 	 * PAM credentials may take the form of supplementary groups.
 	 * These will have been wiped by the above initgroups() call.
-- 
2.14.3 (Apple Git-98)

