diff --git a/ssh_config b/ssh_config
index 4e879cd..9b0a4e4 100644
--- a/ssh_config
+++ b/ssh_config
@@ -39,12 +39,16 @@
 #   IdentityFile ~/.ssh/id_ecdsa
 #   IdentityFile ~/.ssh/id_ed25519
 #   Port 22
-#   Protocol 2
+# GEN005501
+Protocol 2
 #   Cipher 3des
-#   Ciphers aes128-ctr,aes192-ctr,aes256-ctr,arcfour256,arcfour128,aes128-cbc,3des-cbc
-#   MACs hmac-md5,hmac-sha1,umac-64@openssh.com,hmac-ripemd160
+# GEN005510, GEN005511
+Ciphers aes128-ctr,aes192-ctr,aes256-ctr
+# GEN005512
+MACs hmac-sha1,hmac-sha2-256,hmac-sha2-512
 #   EscapeChar ~
-#   Tunnel no
+# GEN005532
+Tunnel no
 #   TunnelDevice any:any
 #   PermitLocalCommand no
 #   VisualHostKey no
