diff -ru openssh-7.1p1.orig/sshd_config openssh-7.1p1/sshd_config
--- openssh-7.1p1.orig/sshd_config	Fri Aug 21 00:49:03 2015
+++ openssh-7.1p1/sshd_config	Wed Sep  2 08:52:58 2015
@@ -41,7 +41,7 @@
 # Authentication:
 
 #LoginGraceTime 2m
-#PermitRootLogin prohibit-password
+PermitRootLogin no
 #StrictModes yes
 #MaxAuthTries 6
 #MaxSessions 10
@@ -103,7 +103,7 @@
 #X11DisplayOffset 10
 #X11UseLocalhost yes
 #PermitTTY yes
-#PrintMotd yes
+PrintMotd no
 #PrintLastLog yes
 #TCPKeepAlive yes
 #UseLogin no
-UsePrivilegeSeparation sandbox		# Default for new installations.
+UsePrivilegeSeparation sandbox # GEN005537
 #PermitUserEnvironment no
-#Compression delayed
+Compression no # GEN005539
 #ClientAliveInterval 0
 #ClientAliveCountMax 3
 #UseDNS no
 #PidFile /var/run/sshd.pid
 #MaxStartups 10:30:100
-#PermitTunnel no
+PermitTunnel no # GEN005531
 #ChrootDirectory none
 #VersionAddendum none
 
 # no default banner path
-#Banner none
+Banner /etc/issue
 
 # override default of no subsystems
-Subsystem	sftp	/usr/libexec/sftp-server
+#Subsystem	sftp	/usr/libexec/sftp-server
 
 # Example of overriding settings on a per-user basis
 #Match User anoncvs
@@ -131,3 +132,6 @@ Subsystem	sftp	/usr/libexec/sftp-server
 #	AllowTcpForwarding no
 #	PermitTTY no
 #	ForceCommand cvs server
+Ciphers aes128-ctr,aes192-ctr,aes256-ctr # GEN005505, GEN005506
+MACs hmac-sha1,hmac-sha2-256,hmac-sha2-512 # GEN005507
+AllowGroups staff # GEN005521
