From 15cdf4c6d4635d87d14975809c0f007f648c85dd Mon Sep 17 00:00:00 2001
From: Pavel Zakharov <pavel.zakharov@delphix.com>
Date: Tue, 19 Jun 2018 13:46:45 -0400
Subject: [PATCH] sshd_config.patch

---
 sshd_config | 28 ++++++++++++++++++----------
 1 file changed, 18 insertions(+), 10 deletions(-)

diff --git a/sshd_config b/sshd_config
index 717f485..4359e76 100644
--- a/sshd_config
+++ b/sshd_config
@@ -15,6 +15,9 @@
 #ListenAddress 0.0.0.0
 #ListenAddress ::
 
+# The default requires explicit activation of protocol 1
+Protocol 2
+
 #HostKey /etc/ssh/ssh_host_rsa_key
 #HostKey /etc/ssh/ssh_host_ecdsa_key
 #HostKey /etc/ssh/ssh_host_ed25519_key
@@ -33,7 +36,7 @@
 # Authentication:
 
 #LoginGraceTime 2m
-#PermitRootLogin prohibit-password
+PermitRootLogin no
 #StrictModes yes
 #MaxAuthTries 6
 #MaxSessions 10
@@ -42,7 +45,7 @@
 
 # The default is to check both .ssh/authorized_keys and .ssh/authorized_keys2
 # but this is overridden so installations will only check .ssh/authorized_keys
-AuthorizedKeysFile	.ssh/authorized_keys
+AuthorizedKeysFile	.ssh/authorized_keys /export/home/cli/.ssh/authorized_keys.%u
 
 #AuthorizedPrincipalsFile none
 
@@ -87,33 +90,35 @@ AuthorizedKeysFile	.ssh/authorized_keys
 # and ChallengeResponseAuthentication to 'no'.
 #UsePAM no
 
-#AllowAgentForwarding yes
-#AllowTcpForwarding yes
+AllowAgentForwarding no
+AllowTcpForwarding no
+AllowStreamLocalForwarding no
 #GatewayPorts no
-#X11Forwarding no
+X11Forwarding no
 #X11DisplayOffset 10
 #X11UseLocalhost yes
 #PermitTTY yes
-#PrintMotd yes
+PrintMotd no
 #PrintLastLog yes
 #TCPKeepAlive yes
 #UseLogin no
+UsePrivilegeSeparation sandbox # GEN005537
 #PermitUserEnvironment no
-#Compression delayed
+Compression no # GEN005539
 #ClientAliveInterval 0
 #ClientAliveCountMax 3
 #UseDNS no
 #PidFile /var/run/sshd.pid
 #MaxStartups 10:30:100
-#PermitTunnel no
+PermitTunnel no # GEN005531
 #ChrootDirectory none
 #VersionAddendum none
 
 # no default banner path
-#Banner none
+Banner /etc/issue
 
 # override default of no subsystems
-Subsystem	sftp	/usr/libexec/sftp-server
+#Subsystem	sftp	/usr/libexec/sftp-server
 
 # Example of overriding settings on a per-user basis
 #Match User anoncvs
@@ -121,3 +126,6 @@ Subsystem	sftp	/usr/libexec/sftp-server
 #	AllowTcpForwarding no
 #	PermitTTY no
 #	ForceCommand cvs server
+Ciphers aes128-ctr,aes192-ctr,aes256-ctr # GEN005505, GEN005506
+MACs hmac-sha1,hmac-sha2-256,hmac-sha2-512 # GEN005507
+AllowGroups staff # GEN005521
-- 
2.14.3 (Apple Git-98)

