From 541c762d844f82a7c28eacf7e3a14ca63c4acdae Mon Sep 17 00:00:00 2001
From: Pavel Zakharov <pavel.zakharov@delphix.com>
Date: Tue, 19 Jun 2018 13:46:45 -0400
Subject: [PATCH] username-rewrite.patch

---
 auth.h         |   3 +-
 auth2-pubkey.c |   2 +-
 auth2.c        |   9 ++--
 monitor.c      | 127 ++++++++++++++++++++++++++++++++++++++++++++++++++++++++-
 4 files changed, 134 insertions(+), 7 deletions(-)

diff --git a/auth.h b/auth.h
index ee104d8..ee88485 100644
--- a/auth.h
+++ b/auth.h
@@ -61,7 +61,8 @@ struct Authctxt {
 	int		 failures;
 	int		 server_caused_failure;
 	int		 force_pwchange;
-	char		*user;		/* username sent by the client */
+	char		*user;		/* possibly rewritten username */
+	char		*orig_user;	/* username sent by the client */
 	char		*service;
 	struct passwd	*pw;		/* set if 'valid' */
 	char		*style;
diff --git a/auth2-pubkey.c b/auth2-pubkey.c
index 7b1a1c6..13b5efb 100644
--- a/auth2-pubkey.c
+++ b/auth2-pubkey.c
@@ -185,7 +185,7 @@ userauth_pubkey(struct ssh *ssh)
 				    __func__, ssh_err(r));
 		}
 		/* reconstruct packet */
-		xasprintf(&userstyle, "%s%s%s", authctxt->user,
+		xasprintf(&userstyle, "%s%s%s", authctxt->orig_user,
 		    authctxt->style ? ":" : "",
 		    authctxt->style ? authctxt->style : "");
 		if ((r = sshbuf_put_u8(b, SSH2_MSG_USERAUTH_REQUEST)) != 0 ||
diff --git a/auth2.c b/auth2.c
index be82b2d..3661707 100644
--- a/auth2.c
+++ b/auth2.c
@@ -233,15 +233,18 @@ input_userauth_request(int type, u_int32_t seq, struct ssh *ssh)
 	if ((style = strchr(user, ':')) != NULL)
 		*style++ = 0;
 
+	authctxt->orig_user = xstrdup(user);
+
 	if (authctxt->attempt++ == 0) {
 		/* setup auth context */
 		authctxt->pw = PRIVSEP(getpwnamallow(user));
-		authctxt->user = xstrdup(user);
 		if (authctxt->pw && strcmp(service, "ssh-connection")==0) {
 			authctxt->valid = 1;
+			authctxt->user = xstrdup(authctxt->pw->pw_name);
 			debug2("%s: setting up authctxt for %s",
 			    __func__, user);
 		} else {
+			authctxt->user = xstrdup(user);
 			/* Invalid user, fake password information */
 			authctxt->pw = fakepw();
 #ifdef SSH_AUDIT_EVENTS
@@ -274,11 +277,11 @@ input_userauth_request(int type, u_int32_t seq, struct ssh *ssh)
 		userauth_banner();
 		if (auth2_setup_methods_lists(authctxt) != 0)
 			packet_disconnect("no authentication methods enabled");
-	} else if (strcmp(user, authctxt->user) != 0 ||
+	} else if (strcmp(user, authctxt->orig_user) != 0 ||
 	    strcmp(service, authctxt->service) != 0) {
 		packet_disconnect("Change of username or service not allowed: "
 		    "(%s,%s) -> (%s,%s)",
-		    authctxt->user, authctxt->service, user, service);
+		    authctxt->orig_user, authctxt->service, user, service);
 	}
 	/* reset state */
 	auth2_challenge_stop(ssh);
diff --git a/monitor.c b/monitor.c
index 435ec39..704b2c5 100644
--- a/monitor.c
+++ b/monitor.c
@@ -45,6 +45,7 @@
 #endif
 #include <stdlib.h>
 #include <string.h>
+#include <strings.h>
 #include <stdarg.h>
 #include <stdio.h>
 #include <unistd.h>
@@ -202,6 +203,10 @@ struct mon_table {
 
 #define MON_PERMIT	0x1000	/* Request is permitted */
 
+#define	DELPHIX_ENABLED_USER_CACHE "/var/delphix/server/etc/enabled-users.txt"
+#define	DELPHIX_SYSTEM_NS "SYSTEM"
+#define	DELPHIX_DOMAIN_NS "DOMAIN"
+
 struct mon_table mon_dispatch_proto20[] = {
 #ifdef WITH_OPENSSL
     {MONITOR_REQ_MODULI, MON_ONCE, mm_answer_moduli},
@@ -774,13 +779,124 @@ mm_answer_sign(int sock, Buffer *m)
 	return (0);
 }
 
+/*
+ * If the user didn't specify a namespace and there is a name conflict, default
+ * to the DOMAIN namespace (see LoginService#getUserType).
+ */
+static const char *
+get_namespace(const char *username)
+{
+	FILE *file;
+	char buf[BUFSIZ];
+	const char *namespace = NULL;
+
+	if ((file = fopen(DELPHIX_ENABLED_USER_CACHE, "r")) == NULL)
+		return (NULL);
+
+	while (fgets(buf, sizeof (buf), file) != NULL) {
+		char *token, *lasts;
+
+		/* Get the username. */
+		if ((token = strtok_r(buf, ":", &lasts)) == NULL)
+			continue;
+
+		/*
+		 * If we haven't found the desired username yet, keep
+		 * searching.
+		 */
+		if (strcmp(token, username) != 0)
+			continue;
+
+		/* Get the namespace. */
+		if ((token = strtok_r(NULL, ":", &lasts)) == NULL)
+			continue;
+
+		if (namespace == NULL) {
+			/* This is the first time we've seen this username. */
+			if (strcmp(token, DELPHIX_SYSTEM_NS) == 0)
+				namespace = DELPHIX_SYSTEM_NS;
+			else if (strcmp(token, DELPHIX_DOMAIN_NS) == 0)
+				namespace = DELPHIX_DOMAIN_NS;
+
+			continue;
+		} else {
+			/* There is a name conflict. */
+			namespace = DELPHIX_DOMAIN_NS;
+
+			break;
+		}
+	}
+
+	(void) fclose(file);
+
+	return (namespace);
+}
+
+static char *
+rewrite_username(const char *input)
+{
+	char buf[256];
+	size_t length;
+	const char *namespace = NULL;
+	char *atsign;
+
+	/* If getpwnam(3C) returns a valid user, we don't do any rewriting. */
+	if (getpwnam(input) != NULL)
+		return (NULL);
+
+	if ((atsign = strchr(input, '@')) != NULL) {
+		/*
+		 * Determine the case-normalized namespace, if the user
+		 * supplied one. If the user supplied an invalid namespace, we
+		 * don't do any rewriting.
+		 */
+		if (strcasecmp(atsign + 1, DELPHIX_SYSTEM_NS) == 0)
+			namespace = DELPHIX_SYSTEM_NS;
+		else if (strcasecmp(atsign + 1, DELPHIX_DOMAIN_NS) == 0)
+			namespace = DELPHIX_DOMAIN_NS;
+		else
+			return (NULL);
+
+		length = (size_t)(atsign - input);
+	} else {
+		length = strlen(input);
+	}
+
+	/* Extract the username from the given input. */
+	if (length + 1 > sizeof (buf))
+		return (NULL);
+	(void) strlcpy(buf, input, length + 1);
+
+	/*
+	 * If the user did not specify a namespace, we attempt to determine the
+	 * namespace here.
+	 */
+	if (namespace == NULL)
+		namespace = get_namespace(input);
+
+	/*
+	 * If we couldn't determine the namespace for this user, we don't do
+	 * any rewriting.
+	 */
+	if (namespace == NULL)
+		return (NULL);
+
+	/* Format the result. */
+	if (strlcat(buf, "@", sizeof (buf)) >= sizeof (buf))
+		return (NULL);
+	if (strlcat(buf, namespace, sizeof (buf)) >= sizeof (buf))
+		return (NULL);
+
+	return (strdup(buf));
+}
+
 /* Retrieves the password entry and also checks if the user is permitted */
 
 int
 mm_answer_pwnamallow(int sock, Buffer *m)
 {
 	struct ssh *ssh = active_state;	/* XXX */
-	char *username;
+	char *username, *rewritten;
 	struct passwd *pwent;
 	int allowed = 0;
 	u_int i;
@@ -791,6 +907,13 @@ mm_answer_pwnamallow(int sock, Buffer *m)
 		fatal("%s: multiple attempts for getpwnam", __func__);
 
 	username = buffer_get_string(m, NULL);
+	authctxt->orig_user = xstrdup(username);
+
+	rewritten = rewrite_username(username);
+	if (rewritten != NULL) {
+		free(username);
+		username = rewritten;
+	}
 
 	pwent = getpwnamallow(username);
 
@@ -1360,7 +1483,7 @@ monitor_valid_userblob(u_char *data, u_int datalen)
 	if (buffer_get_char(&b) != SSH2_MSG_USERAUTH_REQUEST)
 		fail++;
 	cp = buffer_get_cstring(&b, NULL);
-	xasprintf(&userstyle, "%s%s%s", authctxt->user,
+	xasprintf(&userstyle, "%s%s%s", authctxt->orig_user,
 	    authctxt->style ? ":" : "",
 	    authctxt->style ? authctxt->style : "");
 	if (strcmp(userstyle, cp) != 0) {
-- 
2.14.3 (Apple Git-98)

