diff -ruN node-v0.10.28.orig/lib/readline.js node-v0.10.28/lib/readline.js
--- node-v0.10.28.orig/lib/readline.js	2014-05-06 18:39:01.887488642 +0000
+++ node-v0.10.28/lib/readline.js	2014-05-06 18:45:51.318092224 +0000
@@ -92,6 +92,9 @@
   }
 
   function onend() {
+    if (typeof self._line_buffer === 'string' && self._line_buffer.length > 0) {
+      self.emit('line', self._line_buffer);
+    }
     self.close();
   }
 
@@ -119,6 +122,11 @@
 
     // input usually refers to stdin
     input.on('keypress', onkeypress);
+    input.on('end', function inputEnd() {
+      if (typeof self.line === 'string' && self.line.length > 0)
+        self.emit('line', self.line);
+      self.close();
+    });
 
     // Current line
     this.line = '';
