--- apache-ant-1.9.3/src/script/runant.py.orig	Wed Mar  6 03:47:26 2013
+++ apache-ant-1.9.3/src/script/runant.py	Fri Mar 29 19:34:14 2013
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2.6
 # Licensed to the Apache Software Foundation (ASF) under one or more
 #  contributor license agreements.  See the NOTICE file distributed with
 #  this work for additional information regarding copyright ownership.
