diff --git a/src/iperf_client_api.c b/src/iperf_client_api.c
index 27bf1ee..7d78616 100644
--- a/src/iperf_client_api.c
+++ b/src/iperf_client_api.c
@@ -323,15 +323,6 @@ iperf_client_end(struct iperf_test *test)
 }
 
 
-static jmp_buf sigend_jmp_buf;
-
-static void
-sigend_handler(int sig)
-{
-    longjmp(sigend_jmp_buf, 1);
-}
-
-
 int
 iperf_run_client(struct iperf_test * test)
 {
@@ -342,11 +333,6 @@ iperf_run_client(struct iperf_test * test)
     struct timeval* timeout = NULL;
     struct iperf_stream *sp;
 
-    /* Termination signals. */
-    iperf_catch_sigend(sigend_handler);
-    if (setjmp(sigend_jmp_buf))
-	iperf_got_sigend(test);
-
     if (test->affinity != -1)
 	if (iperf_setaffinity(test->affinity) != 0)
 	    return -1;
diff --git a/src/iperf_server_api.c b/src/iperf_server_api.c
index 51f3812..7919c46 100644
--- a/src/iperf_server_api.c
+++ b/src/iperf_server_api.c
@@ -434,15 +434,6 @@ cleanup_server(struct iperf_test *test)
 }
 
 
-static jmp_buf sigend_jmp_buf;
-
-static void
-sigend_handler(int sig)
-{
-    longjmp(sigend_jmp_buf, 1);
-}
-
-
 int
 iperf_run_server(struct iperf_test *test)
 {
@@ -452,11 +443,6 @@ iperf_run_server(struct iperf_test *test)
     struct timeval now;
     struct timeval* timeout;
 
-    /* Termination signals. */
-    iperf_catch_sigend(sigend_handler);
-    if (setjmp(sigend_jmp_buf))
-	iperf_got_sigend(test);
-
     if (test->affinity != -1) 
 	if (iperf_setaffinity(test->affinity) != 0)
 	    return -1;
diff --git a/src/main.c b/src/main.c
index cac74b9..36aff36 100644
--- a/src/main.c
+++ b/src/main.c
@@ -110,12 +110,26 @@ main(int argc, char **argv)
     return 0;
 }
 
+
+static jmp_buf sigend_jmp_buf;
+
+static void
+sigend_handler(int sig)
+{
+    longjmp(sigend_jmp_buf, 1);
+}
+
 /**************************************************************************/
 static int
 run(struct iperf_test *test)
 {
     int consecutive_errors;
 
+    /* Termination signals. */
+    iperf_catch_sigend(sigend_handler);
+    if (setjmp(sigend_jmp_buf))
+	iperf_got_sigend(test);
+
     switch (test->role) {
         case 's':
 	    if (test->daemon) {
@@ -157,5 +171,7 @@ run(struct iperf_test *test)
             break;
     }
 
+    iperf_catch_sigend(SIG_DFL);
+
     return 0;
 }
