--- dmsuite/apache-tomcat-7.0.42/bin/startup.sh	2014-08-03 16:17:00.000000000 -0700
+++ dmsuite/apache-tomcat-7.0.42/bin/startup.sh	2014-08-03 14:55:24.000000000 -0700
@@ -21,8 +21,7 @@
 # $Id: startup.sh 1202062 2011-11-15 06:50:02Z mturk $
 # -----------------------------------------------------------------------------
 APACHE_DIR=`pwd`
-cd ../../java/jre7u25linux64
-JRE_HOME=`pwd`
+JRE_HOME=/opt/jdk/jre
 export JRE_HOME
 cd $APACHE_DIR

--- dmsuite/DMSApplicator/kitchen.sh	2014-08-03 16:18:06.000000000 -0700
+++ dmsuite/DMSApplicator/kitchen.sh	2015-04-01 14:53:14.000000000 -0700
@@ -8,8 +8,7 @@
 BASEDIR=`dirname $0`
 cd $BASEDIR
 DIR=`pwd`
-cd ../../dmsuite/java/jre7u25linux64
-JAVA_HOME=`pwd`
+JAVA_HOME=/opt/jdk/jre
 export JAVA_HOME
 cd - > /dev/null
 
